examplePio_f90 linking errors on Summit with the default XL compiler

This linking issue is only reproducible with the default IBM XL compiler on Summit (not reproducible with GNU or NVHPC compilers).

Steps to reproduce this issue with latest SCORPIO master branch on a Summit login node:

```
module load cmake

git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio

mkdir build
cd build

CC=mpicc CXX=mpiCC FC=mpif90 cmake -Wno-dev \
-DWITH_NETCDF=OFF \
-DPnetCDF_PATH=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/xl-16.1.1-10/parallel-netcdf-1.12.2-pwnetvnxzogwlhhpyeu63dnhxhp66btu \
-DPIO_USE_MALLOC=ON \
-DPIO_ENABLE_EXAMPLES=ON \
..

make
```

Build errors:
```
...
Scanning dependencies of target examplePio_f90
[ 74%] Building Fortran object examples/f03/CMakeFiles/examplePio_f90.dir/examplePio.f90.o
** pioexample   === End of Compilation 1 ===
** main   === End of Compilation 2 ===
1501-510  Compilation successful for file examplePio.f90.
[ 75%] Linking CXX executable examplePio_f90
../../src/gptl/libgptl.a(perf_utils.F90.o): In function `__perf_utils_NMOD_shr_sys_flush':
scorpio/src/gptl/perf_utils.F90:(.text+0xdc): undefined reference to `flush'
/usr/bin/ld: link errors found, deleting executable `examplePio_f90'
make[2]: *** [examples/f03/CMakeFiles/examplePio_f90.dir/build.make:101: examples/f03/examplePio_f90] Error 1
make[1]: *** [CMakeFiles/Makefile2:421: examples/f03/CMakeFiles/examplePio_f90.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

This issue is also observed when building some Fortran unit tests (with PIO_ENABLE_TESTS option turned ON)
```
Scanning dependencies of target pio_init_finalize
[ 24%] Building Fortran object tests/general/CMakeFiles/pio_init_finalize.dir/pio_init_finalize.F90.o
** init_finalize   === End of Compilation 1 ===
** pio_tf_test_driver_   === End of Compilation 2 ===
** pio_tf_test_main_   === End of Compilation 3 ===
1501-510  Compilation successful for file pio_init_finalize.F90.
[ 25%] Linking CXX executable pio_init_finalize
../../src/gptl/libgptl.a(perf_utils.F90.o): In function `__perf_utils_NMOD_shr_sys_flush':
scorpio/src/gptl/perf_utils.F90:(.text+0xdc): undefined reference to `flush'
/usr/bin/ld: link errors found, deleting executable `pio_init_finalize'
make[2]: *** [tests/general/CMakeFiles/pio_init_finalize.dir/build.make:108: tests/general/pio_init_finalize] Error 1
make[1]: *** [CMakeFiles/Makefile2:1460: tests/general/CMakeFiles/pio_init_finalize.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/src/gptl/CMakeLists.txt
+++ b/src/gptl/CMakeLists.txt
@@ -61,6 +61,9 @@ elseif (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
     set ( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch" )
   endif ()
 endif ()
+if (CMAKE_Fortran_COMPILER_ID STREQUAL "XL")
+  set ( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qextname=flush" )
+endif ()
 
 #==============================================================================
 #  DEFINE THE INSTALL

